CodeReferenceActionListener.kt

```diff
@@ -0,0 +1,15 @@
+package co.bito.intellij.codeCompletion.toolwindow
+
+import com.intellij.openapi.editor.RangeMarker
+import co.bito.intellij.codeCompletion.model.InvocationContext
+import co.bito.intellij.codeCompletion.model.SessionContext
+import co.bito.intellij.codeCompletion.popup.UserActionListener
+
+class CodeReferenceActionListener : UserActionListener {
+    override fun afterAccept(states: InvocationContext, sessionContext: SessionContext, rangeMarker: RangeMarker) {
+        val (project, editor) = states.requestContext
+        val manager = CodeReferenceManager.getInstance(project)
+        manager.insertCodeReference(states, sessionContext.selectedIndex)
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/mayanksingh0307/TestRepo/pull/3#issuecomment-2196562790>#0911d8</a><span> - 06/28/2024, 10:10 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The method 'insertCodeReference' is called without checking if 'manager' is null. If 'CodeReferenceManager.getInstance(project)' returns null, this will lead to a NullPointerException.
</div>
<div id="fix">
<b>Fix:</b> Add a null check for 'manager' before calling 'insertCodeReference' and 'addListeners' methods.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -11,6 +11,10 @@
      val manager = CodeReferenceManager.getInstance(project)
 +    if (manager != null) {
      manager.insertCodeReference(states, sessionContext.selectedIndex)
      manager.addListeners(editor)
 +    } else {
 +        // Handle the case where manager is null, possibly log an error or throw an exception
 +    }
```
</div>
</details>
</div>
<br>


</div>